git/git.go

```diff
@@ -37,11 +37,21 @@ func Dir() (string, error) {
 }
 
 func HasFile(segments ...string) bool {
-	dir, err := Dir()
-	if err != nil {
-		return false
+	// The blessed way to resolve paths in git dir, since Git 1.5.0
+	// Needed when working with git worktrees.
+	output, err1 := gitOutput("rev-parse", "-q", "--git-path", filepath.Join(segments...))
```

Comment:
If you could assume git 1.5 you could use An alternative approach would be to take advantage of:

```
% git rev-parse --symbolic-full-name @{upstream}
refs/remotes/retronym/topic/typer-debug-implicits
```

But while you are being backwards compatible, it probably isn't worth using that conditionally.